pkg/asset/agent/agentconfig/agenthosts.go

```diff
@@ -0,0 +1,306 @@
+package agentconfig
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"sigs.k8s.io/yaml"
+
+	aiv1beta1 "github.com/openshift/assisted-service/api/v1beta1"
+	"github.com/openshift/installer/pkg/asset"
+	agentAsset "github.com/openshift/installer/pkg/asset/agent"
+	"github.com/openshift/installer/pkg/types/agent"
+	"github.com/openshift/installer/pkg/types/baremetal"
+	"github.com/openshift/installer/pkg/types/baremetal/validation"
+	"github.com/openshift/installer/pkg/validate"
+)
+
+var (
+	_ asset.WritableAsset = (*AgentHosts)(nil)
+)
+
+const (
+	masterRole string = "master"
+	workerRole string = "worker"
+)
+
+type nmStateInterface struct {
+	Interfaces []struct {
+		MACAddress string `json:"mac-address,omitempty"`
+		Name       string `json:"name,omitempty"`
+	} `yaml:"interfaces,omitempty"`
+}
+
+// AgentHosts generates the hosts information from the AgentConfig and
+// OptionalInstallConfig assets.
+type AgentHosts struct {
+	Hosts        []agent.Host
+	RendezvousIP string
+}
+
+// Name returns a human friendly name.
+func (a *AgentHosts) Name() string {
+	return "Agent Hosts"
+}
+
+// Dependencies returns all of the dependencies directly needed the asset.
+func (a *AgentHosts) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&agentAsset.OptionalInstallConfig{},
+		&AgentConfig{},
+	}
+}
+
+// Generate generates the Hosts data.
+func (a *AgentHosts) Generate(dependencies asset.Parents) error {
+	agentConfig := &AgentConfig{}
+	installConfig := &agentAsset.OptionalInstallConfig{}
+	dependencies.Get(agentConfig, installConfig)
+	var hostsFromInstallConfig = false
+
+	if agentConfig.Config != nil {
+		a.RendezvousIP = agentConfig.Config.RendezvousIP
+		a.Hosts = append(a.Hosts, agentConfig.Config.Hosts...)
+	}
+
+	if installConfig != nil && installConfig.Config != nil {
+		if installConfig.Config.Platform.Name() == baremetal.Name && len(installConfig.Config.Platform.BareMetal.Hosts) > 0 {
+			fieldPath := field.NewPath("Platform", "Baremetal", "Hosts")
+			if len(a.Hosts) == 0 {
+				logrus.Debugf("Using %s from %s", fieldPath, agentAsset.InstallConfigFilename)
```

Comment:
It seems to me that by setting the log trace as `Info` (which could be a useful hint for the user), then the `hostsFromInstallConfig` may not be required (it would be clear from where an eventual error will come from)

Reply:
Good idea. Changed to use Info and to remove the use of `hostsFromInstallConfig`